compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -777,9 +777,7 @@ impl Diagnostic {
         applicability: Applicability,
         style: SuggestionStyle,
     ) -> &mut Self {
-        let mut suggestions: Vec<_> = suggestions.into_iter().collect();
-        suggestions.sort();
-
+        let suggestions: Vec<_> = suggestions.into_iter().collect();
```

Comment:
This collect is not needed anymore. Please remove it and move the debug assert after the substitutions collect.